Update Add/Edit Shipping Time Modals to support minimum and maximum shipping times stepper

jorgemd24 opened the pull request:
### Changes proposed in this Pull Request:

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

Part of https://github.com/woocommerce/google-listings-and-ads/pull/2594

Since we're now working with min and max shipping times, the "Edit" and "Add" modals need to be updated to use the stepper component when creating or editing shipping times. This PR introduces a new component, `MinMaxShippingTimes`, which includes steppers for both the minimum and maximum shipping times.

![image](https://github.com/user-attachments/assets/d70ec3c0-62b9-42ed-b286-bb04e02d2b01)

![image](https://github.com/user-attachments/assets/3d930dc7-044c-4454-9719-ed35fa105f21)

The errors in the modals are displayed within the form instead of beneath the stepper component to prevent issues like this:

![image](https://github.com/user-attachments/assets/df1e8059-6341-48bf-b3ad-481bc4e05390)




### Screenshots:

<!--- Optional --->


### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

1. Follow the steps mentioned in this PR to update the database: https://github.com/woocommerce/google-listings-and-ads/pull/2520
2. Navigate to Dashboard -> Free Listings -> Edit
3. Add and edit shipping times using the modals, and verify that the min and max times are updated correctly.


### Additional details:

In this commit [8652d24](https://github.com/woocommerce/google-listings-and-ads/pull/2609/commits/8652d2421db60f69107ae69c8c6355440493dbfa) I updated the `onBlur` function to align with `onIncrement`. However, I'm unsure what's best—whether we should allow users to enter any value and display errors in the form, or restrict input to values within the allowed range, like preventing negative numbers. Any advice would be appreciated.

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with change type prefix`(Fix|Add|…) - `, for example:
> Break - A change breaking previous API or functionality.
> Add - A new feature, function or functionality was added.
> Update - Big changes to something that wasn't broken.
> Fix - Took care of something that wasn't working.
> Tweak - Small change, that isn't actually very important.
> Dev - Developer-facing only change.
> Doc - Updated customer or developer facing documentation

If you remove the "Changelog entry" header, the Pull Request title will be used as the changelog entry.

Add the `changelog: none` label if no changelog entry is needed.
-->
### Changelog entry

>

jorgemd24 committed:
Add stepper to add time button

jorgemd24 committed:
Add edit time stepper in appmodal

jorgemd24 committed:
Add css

jorgemd24 committed:
Merge branch 'add/min-max-time-inputs' into update/add-and-edit-shipping-time-inputs

jorgemd24 committed:
Create MinMaxShippingTimes component

jorgemd24 committed:
Refactor add and edit modals to use the min max component

jorgemd24 committed:
simplify handleblur

jorgemd24 committed:
Refactor countries time input to use the MinMax shipping times

jorgemd24 committed:
Adjust styling

jorgemd24 committed:
Add tests for AddTimeModal

jorgemd24 committed:
Add tests for EditTimeModal

jorgemd24 committed:
Fix linting

jorgemd24 committed:
Show errors in add time modal

jorgemd24 committed:
Show errors in edit modal

jorgemd24 committed:
Adjust validateShippingTime

jorgemd24 committed:
Only update the value if the shipping time is between the min and max allowed

martynmjones reviewed (commented):
Hey @jorgemd24, thanks for implementing the stepper for shipping.

All is working properly although I noticed a few useability issues which we may want to address before releasing.

1. The modal size changes when an error is displayed or hidden. If a merchant is using the `+` and `-` to set the value then the content jumps and the mouse may move off the button when the error status changes.

https://github.com/user-attachments/assets/c5047242-94fe-4c11-b117-f421739d7e68

2. Invalid fields aren't highlighted when attempting to save changes. If a merchant is shipping to a number of countries and has different settings for each then it would be quite difficult to spot where the issue is.

<img width="1166" alt="Screenshot 2024-09-19 at 11 22 05" src="https://github.com/user-attachments/assets/bdcc1128-952a-4f9a-9caf-4ad95c6c7023">

3. Locations are immediately grouped if the min/max values match an existing field. at the moment merchants would need to enter an identical value and then focus on a different element before they are grouped.

    As it happens when using the steppers we potentially have a situation where we want to use the `+` button to get to `4` but it's grouped before that can happen.

    What do you think of adding a small delay before grouping so that it would be possible to use the stepper to get to a higher number?

https://github.com/user-attachments/assets/75c6a3fa-8c69-4932-9fe8-986b366f45bd

jorgemd24 committed:
Fix issue with variable height in edit and add modal

jorgemd24 committed:
Add delay before saving

jorgemd24 committed:
Adjust tests

jorgemd24 commented:
Thanks @martynmjones for your helpful comments! I’ve addressed points 1 and 3.

Regards the second one: 

> Invalid fields aren't highlighted when attempting to save changes. If a merchant is shipping to a number of countries and has different settings for each then it would be quite difficult to spot where the issue is.

I agree that it's not easy to see which one is incorrect, but this behaviour hasn't changed compared to the one in the `develop` branch. 

![image](https://github.com/user-attachments/assets/1de57f06-eb22-4350-8bb8-2b4d9f5dec8c)

I believe the issue is that we're treating `shipping_times` as a group of inputs rather than individual inputs, which makes it harder to display errors for each input separately. I'll create an issue for this since I think it goes beyond the scope of this PR.

martynmjones reviewed (approved):
Many thanks for making those adjustments @jorgemd24! Tested the updated version and both the issues I mentioned previously have been resolved. 

> I believe the issue is that we're treating shipping_times as a group of inputs rather than individual inputs, which makes it harder to display errors for each input separately. I'll create an issue for this since I think it goes beyond the scope of this PR.

Perfect! Thank you.
